xapian-applications/omega/query.cc

```diff
@@ -752,11 +756,13 @@ html_highlight(const string &s, const string &list,
 	}
 	j = term_end;
 	term = Xapian::Unicode::tolower(term);
-	int match = word_in_list(term, list);
+	build_word_map(list);
```

Comment:
Yes, I agree. Defining a new class and using that would be a clean solution. I'm working on it now; will update the PR shortly.